### PR TITLE
[REFACTOR] Simplify activity & configuration mapper generation

### PR DESCRIPTION
**Pull request by pcrosemurgy**
* **Tickets addressed:** N/A
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

- Uses fully qualified configuration mapper class in generated code to avoid any possible name collisions.
- Simplifies `MapperMethodMaker`, removes parsing logic from this class (for example, `getParameters()`).

## Verification

These changes did not result in unwanted changes to generated code. To demonstrate:

- Run `gradle clean classes` on `develop` and this branch, saving `examples/banananation/build/generated` with `cp -r <some_dir>`.
- Run `diff -qr banana_gen refactor_gen` (the saved directories from above)
    ```
    Files .../GeneratedMissionModelFactory.java and .../GeneratedMissionModelFactory.java differ
    ```
- Run `diff .../GeneratedMissionModelFactory.java .../GeneratedMissionModelFactory.java` (with full paths substituted):
    ```diff
    @@ -19,7 +19,7 @@
     public final class GeneratedMissionModelFactory implements MissionModelFactory<RootModel<Mission>> {
       @Override
       public Optional<ConfigurationType<?>> getConfigurationType() {
    -    return Optional.of(new ConfigurationMapper());
    +    return Optional.of(new gov.nasa.jpl.aerie.banananation.generated.ConfigurationMapper());
       }
    
       @Override
    @@ -35,7 +35,7 @@
    
         try {
           final var serializedArguments = configuration.asMap().orElseThrow(() -> new     ConfigurationType.UnconstructableConfigurationException());
    -      final var deserializedConfig = new ConfigurationMapper().instantiate(serializedArguments);
    +      final var deserializedConfig = new gov.nasa.jpl.aerie.banananation.generated.ConfigurationMapper().instantiate(serializedArguments);
           final var model = InitializationContext.initializing(executor, builder, () -> new Mission(registrar, deserializedConfig));
           return new RootModel<Mission>(model, executor);
         } catch (final ConfigurationType.UnconstructableConfigurationException ex) {
    ```

## Documentation
Updated javadocs.

## Future work
None.

### CODE REVIEW

**Review by patkenneally — APPROVED** [review]

File: merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java [hunk 1]
```diff
@@ -4,48 +4,48 @@
...
 import java.util.stream.Collectors;
 
-public interface MapperMethodMaker {
+public abstract sealed class MapperMethodMaker permits
+    AllDefinedMethodMaker,
+    AllStaticallyDefinedMethodMaker,
+    NoneDefinedMethodMaker,
+    SomeStaticallyDefinedMethodMaker
+{
+  protected final ExportTypeRecord exportType;
```
Comment (mattdailis) [inline merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java]: Just so I can understand - protected means the field is accessible by any class within this package, as well as by subclasses of this class in other packages?

If all of the sealed subclasses are in this package, would it make sense for these to be package private? (Happy to hear a counter-argument - I haven't used `protected` much before)

File: merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java [hunk 2]
```diff
@@ -119,8 +118,24 @@ default MethodSpec makeGetArgumentsMethod(final ExportTypeRecord exportType) {
         .build();
   }
 
-  default MethodSpec makeGetValidationFailures(final ExportTypeRecord exportType) {
-    final var metaName = getMetaName(exportType);
+  public final MethodSpec makeGetRequiredParametersMethod() {
```
Comment (mattdailis) [inline merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java:125]: I like the explicit final/not-final demarcations guiding the subclasses as to which methods to actually implement :+1:

File: merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java (hunk 1 above)
Comment (pcrosemurgy) [reply merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java]: Ahh true, I always forget that `protected` is less strict than package private. I'll make this package private :)

**Review by mattdailis — APPROVED** [review]
Definitely steps in the right direction 👍 Looks good to me.

File: merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java (hunk 1 above)
Comment (pcrosemurgy) [reply merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java]: Just pushed change for this

**Pull request merged**